src/applications/disability-benefits/2346/containers/OrderTable.jsx

```diff
@@ -0,0 +1,186 @@
+        };
+      });
+    })();
+
+    const UITableHeaders = Object.keys(UIDataObjs[0]).map(UIDataObjKey => {
+      switch (UIDataObjKey) {
+        case 'dateProcessed':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Date Processed
+            </th>
+          );
+        case 'product':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Model
+            </th>
+          );
+        case 'id':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Serial Number
+            </th>
+          );
+        case 'orderIntakeMethod':
+          return (
+            <th className="form2346 vads-c-table__th" key={UIDataObjKey}>
+              Order Method
+            </th>
+          );
+        case 'checkbox':
+          return (
+            <th
+              className="form2346 vads-c-table__th--narrow vads-u-text-align--center"
+              key={UIDataObjKey}
+            >
+              Check to Order Batteries
+            </th>
+          );
+        default:
+      }
+      return UIDataObjKey;
+    });
+
+    const UITableRowsAndData = UIDataObjs.map(UIDataObj => {
+      const UIDataObjValues = Object.values(UIDataObj);
+
+      return (
+        <tr key={UIDataObj.id}>
+          {UIDataObjValues.map(UIDataObjValue => (
+            <td key={UIDataObjValue}>{UIDataObjValue}</td>
+          ))}
+        </tr>
+      );
+    });
+
+    return {
+      UITableHeaders,
+      UITableRowsAndData,
+    };
+  }
+
+  render() {
+    const { firstName, lastName, data, address, city, state, zip } = this.props;
+    let dlcTableHeaders;
+    let dlcTableRowsAndData;
+    // NOTE: Only run the following code when the DLC's data
+    // has been received to improve performance
+    if (data.length > 0) {
+      const { UITableHeaders, UITableRowsAndData } = this.createDlcUIData(data);
+      dlcTableHeaders = UITableHeaders;
+      dlcTableRowsAndData = UITableRowsAndData;
+    }
+    return (
+      <div className="vads-c-form2346">
+        {data.length > 0 && (
+          <>
+            <h2>
+              <strong>Name:</strong> {firstName} {lastName}
+            </h2>
+            <h2>
+              <strong>Address:</strong> {address} <br /> {city} {state} {zip}{' '}
+            </h2>
+            <table className="form2346 vads-c-table">
+              <tbody>
+                <tr>{dlcTableHeaders}</tr>
+                {dlcTableRowsAndData}
+              </tbody>
+            </table>
+            <button
+              type="button"
+              className="usa-button-primary va-button-primary"
+            >
+              Submit Your Order
+            </button>
+          </>
+        )}
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  data: state.form2346Reducer.data == null ? '' : state.form2346Reducer.data,
+  firstName:
```

Comment:
the initialState for these fields could be set to ```''``` and then you should use optional chaining to access them:

```firstName: state?.profile?.firstName```

Reply:
@bradl3yC @kelsonic Is this what you mean?

`const mapStateToProps = state => ({
  data: '' || state?.form2346Reducer?.data,
  firstName: '' || state?.user?.profile?.userFullName?.first,
  lastName: '' || state?.user?.profile?.userFullName?.last,
  address: '' || state?.user?.profile?.vet360?.mailingAddress?.addressLine1,
  city: '' || state?.user?.profile?.vet360?.mailingAddress?.city,
  state: '' || state?.user?.profile?.vet360?.mailingAddress?.stateCode,
  zip: '' || state?.user?.profile?.vet360?.mailingAddress?.zipCode,
});`

Reply:
@mr0sari0 Hmm, so that's saying `false || state?.form2346Reducer?.data`, which will always equate to `state?.form2346Reducer?.data`.

I think you'll want to write it similar to this if you want default values:

```jsx
const mapStateToProps = state => ({
  address: state?.user?.profile?.vet360?.mailingAddress?.addressLine1 || '',
  city: state?.user?.profile?.vet360?.mailingAddress?.city || '',
  data: state?.form2346Reducer?.data || '',
  firstName: state?.user?.profile?.userFullName?.first || '',
  lastName: state?.user?.profile?.userFullName?.last || '',
  state: state?.user?.profile?.vet360?.mailingAddress?.stateCode || '',
  zip: state?.user?.profile?.vet360?.mailingAddress?.zipCode || '',
});
```

Alternatively, you can set default values for those props using `static defaultProps` 🙂